src/stdatamodels/schema.py

```diff
@@ -169,11 +168,17 @@ def merge_property_trees(schema):
     can be represented in individual files and then referenced elsewhere. They
     are then combined by this function into a single schema data structure.
     """
-    newschema = OrderedDict()
+    # track the "combined" and "top" items separately
+    # this allows the top level "id", "$schema", etc to overwrite
+    # combined items (so that the schema["id"] doesn't change).
+    combined_items = {}
+    top_items = {}
 
     def add_entry(path, schema, combiner):
+        if not top_items:
+            top_items.update(schema)
         # TODO: Simplify?
```

Comment:
I'd say we leave it. The code below it could certainly use some simplification. Although the comment is at least [8 years old](https://github.com/spacetelescope/jwst/commit/5915b0e919be1e3393fd0ba74ff9106de77a4a1e).

Reply:
I opened an issue to increase the visibility of this: https://github.com/spacetelescope/stdatamodels/issues/366